doc/dev/adr/0023-property-support.md

```diff
@@ -0,0 +1,306 @@
+  "@id": "dtmi:propertySketch:PropertySketch;1",
+  "@type": [ "Interface", "Mqtt" ],
+  "payloadFormat": "Json/ecma/404",
+  "propertyTopic": "sample/property/{propertyName}/{action}",
+  "contents": [
+    {
+      "@type": "Property",
+      "name": "Foo",
+      "schema": "integer"
+    },
+    {
+      "@type": "Property",
+      "name": "Bar",
+      "writable": true,
+      "schema": "string"
+    }
+  ]
+}
+```
+
+This model generates three Commands:
+
+| Command name | MQTT topic | Example request | Example response |
+| --- | --- | --- | --- |
+| ReadFoo | "sample/property/Foo/read" | | { "Foo": 33 } |
+| ReadBar | "sample/property/Bar/read" | | { "Bar": "hello" } |
+| WriteBar | "sample/property/Bar/write" | { "Bar": "bye" } | |
+
+The two read Commands have no request payloads.
+Their response payloads are objects, each with a single required field.
+The reverse holds for the write Command, which is defined only for the writable property.
+
+#### Basic sample model with aggregated Property topics
+
+The next sample model is identical to the above except for the property topic value, which does not contain a "{propertyName}" token:
+
+```json
+  "propertyTopic": "sample/property/{action}"
+```
+
+This model generates two Commands:
+
+| Command name | MQTT topic | Example request | Example response |
+| --- | --- | --- | --- |
+| Read | "sample/property/read" | | { "Foo": 33, "Bar": "hello" } |
+| Write | "sample/property/write" | { "Bar": "bye" } | |
+
+In general, a model that does not contain a "{propertyName}" token generates at most two Commands, one for reading and one for writing.
+If no Property in the Interface is writable, only a read Command is generated.
+
+The read Command has no request payload, and the write Command has no response payload.
+Fields in the read response are not optional and not nullable; all values are read on each read Command.
+Fields in the write request are optional; only fields that are present and have non-null values will be written.
+
+### New adjunct types: PropertyResult, PropertyValue, ReadError, and WriteError
+
+The second additive change to the Mqtt extension is support for modeling the error conditions that can result from a Property read or write.
+Mechanisms for modeling errors in Command responses are defined in [ADR 19](./0019-codegen-user-errs.md).
+The present ADR adds closely analogous mechanims for modeling errors in Property read/write responses.
+Specifically, the following four adjunct types are added.
+
+| New Adjunct Type | Material Cotype | Analogous ADR19 Type |
+| --- | --- | --- |
+| PropertyResult | Object | Result |
+| PropertyValue | Field | NormalResult |
+| ReadError | Field | ErrorResult |
+| WriteError | Field | ErrorResult |
+
+These adjunct types behave nearly identically, *mutatis mutandis*, to the analogous types defined for Command in ADR19.
+Specifically:
+
+* PropertyResult defines the type sent on the wire in a read response.
```

Comment:
PropertyResult/PropertyValue were a bit hard for me to link properly to read vs write - can you add the takeaways from these sentences to the table? I think that would help me understand this much better!
ex. extra column specifying whether that line corresponds to the read response with no error/write response, etc

I was thinking about WriteResult/ReadResult to align more with both the Error names and the Command adjunct types, but it looks like that would be too specific. (although maybe ReadErrorResult/WriteErrorResult would add better symmetry)

Reply:
I'll take a shot at modifying the table and see how it comes out.

Regarding names, I kind of wanted to avoid the term "result" since we're not supporting arbitrary functions that return results.  It is the Write action or the Read action that can yield an error.

Reply:
Added a column to the table.